src/api/Base.js

```diff
@@ -123,6 +129,77 @@ class Base {
     getCache(): Cache {
         return this.cache;
     }
+
+    /**
+     * Generic success handler
+     *
+     * @param {Object} data the response data
+     * @param {Function} successCallback the success callback
+     */
+    successHandler(data: Object, successCallback: Function) {
+        if (!this.isDestroyed()) {
+            successCallback(data);
+        }
+    }
+
+    /**
+     * Generic error handler
+     *
+     * @param {Object} data the response data
+     * @param {Function} errorCallback the error callback
+     */
+    errorHandler(error: Object): void {
+        if (!this.isDestroyed() && this.errorCallback) {
```

Comment:
Might as well check if it's a function if you are checking for it's existence?

Reply:
I think flow will take care of this

Reply:
Only statically, probably should add this https://github.com/brigand/babel-plugin-flow-react-proptypes

Reply:
Will add the function type check